### DISCUSSION TITLE
What all does pool.SAGPooling return?

### DISCUSSION THREAD

**Discussion by Sar2580P** [pyg-team/pytorch_geometric#9039] [General]
There is no hint at what does sag_pool return in the docs , through code_base I only know that it return tuple ... But  what that tuple exactly composed of... 

Here is the docs I'm referring to --->
https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.pool.SAGPooling.html

**Comment by rusty1s** [comment]
Sorry that the documentation isn't really helpful in this case :(

You can browse the code to see what is returned exactly:
* The pooled node features
* The coarsened graph representation `edge_index`
* The filtered edge features (if any)
* The filtered `batch` object
* The node indices which were preserved in the coarsened graph
* The attention scores that of preserved nodes
